packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -79,6 +79,7 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
       absorb(oldPosition);
     if (keepScrollOffset)
       restoreScrollOffset();
+    addListener(_updateSemanticActions);
```

Comment:
Rather than listening to yourself, you can just override the method that calls the listeners.